app/controllers/application_controller/report_downloads.rb

```diff
@@ -83,14 +89,28 @@ def render_report_data
   # Send rendered report data
   def send_report_data
     if @sb[:render_rr_id]
-      rr = MiqReportResult.find(@sb[:render_rr_id])
-      filename = filename_timestamp(rr.report.title, 'export_filename')
       disable_client_cache
-      generated_result = rr.get_generated_result(@sb[:render_type])
-      rr.destroy
-      send_data(generated_result,
-                :filename => "#{filename}.#{@sb[:render_type]}",
-                :type     => "application/#{@sb[:render_type]}")
+      @result = MiqReportResult.find(@sb[:render_rr_id])
+      @report = @result.report
+      @data = @result.get_generated_result(@sb[:render_type])
+      # We need the last_run_on time from the original result
+      last_run_on = MiqReportResult.find(session[:report_result_id]).last_run_on
```

Comment:
does this fetch the whole record and then take a single columnt or does it fetch just the single column from the db?

Reply:
nice catch, fixed